Intersect.Server/Entities/Npc.cs

```diff
@@ -1633,6 +1633,38 @@ public override EntityPacket EntityPacket(EntityPacket packet = null, Player for
             return pkt;
         }
 
+        public override void OnNearEntityDies(Entity en)
+        {
+            if (en.GetEntityType() == EntityType.Player)
+            {
+                RemoveFromDamageMap(en);
+            }
+        }
+
+        public override void OnNearEntityAttacked(Entity victim, Entity attacker) 
+        {
+            if (Target != null)
+            {
+                return;
+            }
+
+            if (!IsAllyOf(victim))
+            {
+                return;
+            }
+
+            if (!InRangeOf(attacker, Base.SightRange))
+            {
+                return;
+            }
+
+            if (victim.GetEntityType() != EntityType.Player)
+            {
+                return;
+            }
```

Comment:
This logic does not make sense.

You refactored Player.NotifySwarm(), but you didn't refactor the Npc.NotifySwarm()? They are essentially the same, the difference being what qualifies as an "ally".

I also think some of the current code is wrong.

Reply:
The `Target` check stays here.

I suggest refactoring out the `IsAllyOf` and `InRangeOf` checks into the base class.

The entity type check is already part of Npc's `IsAllyOf` method, and so this is redundant.